src/model/sponsored_message.rs

```diff
@@ -67,16 +67,20 @@ impl SponsoredMessage {
 
         // TODO: Support multiple sponsored messages
         if let Some(sponsored_message) = sponsored_messages.messages.first() {
-            Ok(Some(Self::new(
-                sponsored_message,
-                &session.chat(sponsored_message.sponsor_chat_id),
-            )))
+            let sponsor_label = match &sponsored_message.sponsor.r#type {
+                tdlib::enums::MessageSponsorType::Bot(sponsor) => {
+                    session.user(sponsor.bot_user_id).first_name()
+                }
+                tdlib::enums::MessageSponsorType::PublicChannel(sponsor) => {
+                    session.chat(sponsor.chat_id).title()
+                }
+                tdlib::enums::MessageSponsorType::PrivateChannel(sponsor) => sponsor.title.clone(),
+                tdlib::enums::MessageSponsorType::Website(sponsor) => sponsor.name.clone(),
+            };
```

Comment:
I think it's much better to move this `match` into the `new` function.  
In the end I'd like having something like this 

```rust
impl SponsoredMessage {
    fn new(
        sponsored_message: tdlib::types::SponsoredMessage,
        session: &model::ClientStateSession,
    ) -> Self {
        use tdlib::enums::MessageSponsorType::*;

        glib::Object::builder()
            .property("message-id", sponsored_message.message_id)
            .property(
                "content",
                model::BoxedMessageContent(sponsored_message.clone().content),
            )
            .property(
                "sponsor-label",
                match sponsored_message.sponsor.r#type {
                    Bot(sponsor) => session.user(sponsor.bot_user_id).first_name(),
                    PublicChannel(sponsor) => session.chat(sponsor.chat_id).title(),
                    PrivateChannel(sponsor) => sponsor.title,
                    Website(sponsor) => sponsor.name,
                },
            )
            .build()
    }

    pub(crate) async fn request(
        chat_id: i64,
        session: &model::ClientStateSession,
    ) -> Result<Option<Self>, tdlib::types::Error> {
        tdlib::functions::get_chat_sponsored_messages(chat_id, session.client_().id())
            .await
            .map(
                |tdlib::enums::SponsoredMessages::SponsoredMessages(mut sponsored_messages)| {
                    sponsored_messages
                        .messages
                        .pop()
                        .map(|sponsored_message| Self::new(sponsored_message, session))
                },
            )
    }
}
```

Reply:
I also thought about that approach, but I didn't want to pass `session` to the `new` function (I like keeping the same arguments). I'll do it.

Reply:
The point is, you wouldn't set a different `sponsor_label` than the `match sponsored_message.sponsor.r#type` produces. So, it makes sense to keep this sponsor_label creation label as local as possible.